src/core/orderbookutils.js

```diff
@@ -71,7 +109,16 @@ OrderBookUtils.getOfferQuality = function(offer, currencyGets) {
       reference_date: new Date()
     });
   } else {
-    amount = Amount.from_json(offer.quality + IOU_SUFFIX);
+
+    let currency = offer.TakerGets.currency;
+    let issuer = offer.TakerGets.issuer;
+
+    if (currency === undefined || issuer === undefined) {
+      currency = offer.TakerPays.currency;
```

Comment:
Well the initial purpose of this task was to eliminate the arbitrary values in IOU_SUFFIX.

`quality` is a ratio of amount values, so `getOfferQuality` should return an `IOUValue` rather than an `Amount`.